client/ayon_hiero/api/plugin.py

```diff
@@ -818,12 +823,15 @@ def get(key):
         }
 
         # build product name from layer name
-        if self.base_product_name == "<track_name>":
-            self.base_product_name = self.track_name
+        if self.base_product_variant == "<track_name>":
+            self.base_product_variant = self.track_name
+            self.variant = self.track_name
+        else:
+            self.variant = self.base_product_variant
 
         # create product for publishing
         self.product_name = (
-            f"{self.product_type}{self.base_product_name.capitalize()}")
+            f"{self.product_type}{self.base_product_variant.capitalize()}")
```

Comment:
Shouldn't this be `self.variant`? (why we don't use product name logic bsed on settings?)
```suggestion
            f"{self.product_type}{self.variant.capitalize()}")
```

Reply:
Issue here is that we cannot use the editorial task in the templated product name creation.

Reply:
improved via aa5bea3fa8a831641428a2206aee899e69f754e8